### PR TITLE
[MAD PP] José María Lora & Paula García

### PR DESCRIPTION
**Pull request by paulaga**
- Hemos conseguido hasta la Iteración 2 y encaminados a la 3
- Muy complicado, complicadísimo, hemos entrado en never ending for loop...
- Comentario: Intentaremos tratar de avanzar :-S

### CODE REVIEW

File: starter_code/src/movies.js [hunk 1]
```diff
@@ -1,12 +1,37 @@
...
+};
 
 // Get the average of all rates with 2 decimals 
-
+function ratesAverage(arr){ 
+  var total = arr.reduce(function(accumulator, current){
+    return accumulator + parseInt(current.rate) / arr.length;
+  },0);
+  return Math.round(total * 100)/100;  
+};
```
Comment (ta-web-mad) [inline starter_code/src/movies.js]: function ratesAverage(arr) {
  var averageRate = arr.reduce(function (acc, elem) {
    if (!elem.rate) return acc;
    var rateFloat = parseFloat(elem.rate).toFixed(2);
    return (rateFloat / arr.length) + acc;
  }, 0);
  return averageRate;
}

File: starter_code/src/movies.js [hunk 2]
```diff
@@ -1,12 +1,37 @@
...
-
+function dramaMoviesRate(arr){
+  for(var i = 0; i < arr.length; i++){
+    arr.filter(function(e){
+      console.log(e.genre);
+      return(e.genre);
+    });
+  }
+ 
+}
```
Comment (ta-web-mad) [inline starter_code/src/movies.js]: function dramaMoviesRate(arr) {
  var drama = arr.filter(function (elem) {
    return elem.genre.includes('Drama');
  });
  if (!drama.length) return;
  var averageDrama = ratesAverage(drama).toFixed(2);
  return parseFloat(averageDrama);
}

File: starter_code/src/movies.js [hunk 3]
```diff
@@ -1,12 +1,37 @@
...
 
 // Order by time duration, in growing order
```
Comment (ta-web-mad) [inline starter_code/src/movies.js]: // Order by time duration, in growing order
function orderByDuration(arr) {
  arr.sort(function (a, b) {
    if (a.duration === b.duration) {
      if (a.title > b.title) {
        return 1;
      }
      if (a.title < b.title) {
        return -1;
      }
      return 0;
    }
    return a.duration - b.duration;
  });
  return arr;
}

// How many movies did STEVEN SPIELBERG
function howManyMovies(arr) {
  if (arr.length === 0) return;
  var drama = arr.filter(function (elem) {
    return elem.genre.includes('Drama');
  });
  var spielbergDramas = drama.filter(function (elem) {
    return elem.director.includes('Spielberg');
  });
  return 'Steven Spielberg directed ' + spielbergDramas.length + ' drama movies!';
}

// Order by title and print the first 20 titles
function orderAlphabetically(arr) {
  var result = [];
  var orderedArr = arr.sort(function (a, b) {
    if (a.title > b.title) {
      return 1;
    }
    if (a.title < b.title) {
      return -1;
    }
    return 0;
  });
  var limit = 20;
  if (orderedArr.length < 20) {
    limit = orderedArr.length;
  }
  for (var i = 0; i < limit; i++) {
    result.push(orderedArr[i].title);
  }
  return result;
}
// console.log(orderAlphabetically(movies));

// Best yearly rate average

function bestYearAvg(movies) {
  if (movies.length === 0) { return undefined; }

  var disctintYears = [];
  var aux = [];

  movies.forEach(function (e) {
    if (!aux.includes(e.year)) {
      aux.push(e.year);
      disctintYears.push({ year: e.year, rate: 0, films: 0 });
    }
  });

  disctintYears.forEach(function (y) {
    movies.forEach(function (e) {
      if (e.year === y.year) {
        if (!e.rate) {
          return;
        }
        y.rate += parseFloat(e.rate);
        y.films += 1;
      }
    });
    y.rate = parseFloat(y.rate / y.films);
  });

  disctintYears.sort(function (a, b) {
    if (a.rate === b.rate) {
      return a.year - b.year;
    }
    return b.rate - a.rate;
  });

  console.log(disctintYears);

  return 'The best year was ' + disctintYears[0].year + ' with an average rate of ' + disctintYears[0].rate;
}

**Comment by ta-web-mad** [comment]
Si teneis un ratillo, echadle un ojo a los comentarios al código, con las soluciones.
Este ejercicio era complicadillo, ya lo sé...

Susana

File: starter_code/src/movies.js (hunk 3 above)
Comment (paulaga) [reply starter_code/src/movies.js]: Muchas gracias Susana :-)

**Pull request closed without merging**